src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -63,6 +68,7 @@ public string[] LiteralPath
         /// </summary>
         [Parameter(ValueFromPipelineByPropertyName = true)]
         [Alias("Type")]
+        [ArgumentCompleter(typeof(PropertyTypeArgumentCompleter))]
```

Comment:
What if it is not Registry provider?

Reply:
From documentation: _This is a dynamic parameter made available by the Registry provider. The Registry provider and this parameter are only available on Windows._

So my understanding is this parameter only works with registry provider. I honestly have seen it not used with any other provider 😄.
 
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/new-itemproperty?view=powershell-7.3#-propertytype

Reply:
It is not correct. The parameter is a _regular_ parameter which is used by provider as generic parameter. The provider can be any custom provider.

Reply:
Can you share an example? I have never seen it used with custom provider and didn't think it was even possible without RegistryProvider.

Reply:
PowerShell offers a public API for creating custom providers of various kinds. And provider cmdlets should support this and work correctly with any custom provider. 
You can find examples on GitHub by searching `IPropertyCmdletProvider`.